release_dashboard/lib/logic/repositories_str.dart

```diff
@@ -0,0 +1,26 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../models/repositories.dart';
+
+const String kEngineStr = 'engine';
+const String kFrameworkStr = 'framework';
+
+/// Helper function that returns the string 'engine' or 'framework' based on the [Repositories] enum.
+///
+/// Also supports returning first letter capitalized string.
+String repositoriesStr(Repositories repository, [bool? capitalized = false]) {
```

Comment:
Can we override `toString` in `Repositories` to handle this logic?

Reply:
You can get temporarily unblocked by adding an extension method here, than when the upstream patch is available remove the extension method.

Reply:
Which `Repositories` you are referring to? The 'Repository` class in the core? Or the enum in the release dashboard called `Repositories`

Reply:
I just added an extension method here, please take another look.

Reply:
I also created a todo to have a method in the core